### ISSUE TITLE
TensorFlow 1.11.0 requires scipy

### ISSUE THREAD

**Issue by drasmuss** [tensorflow/tensorflow#22573]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: n/a
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.11.0
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: n/a
- **GCC/Compiler version (if compiling from source)**: n/a
- **CUDA/cuDNN version**: n/a
- **GPU model and memory**: n/a
- **Exact command to reproduce**: `pip install tensorflow==1.11.0`

### Describe the problem

TensorFlow 1.11.0 requires `keras-preprocessing>=1.0.3`, which in turn requires `scipy>=0.14`.  So now scipy is a dependency of TensorFlow.  This seems like a rather large change, and I didn't see anything about it in the release notes, so I'm wondering whether it was intentional or not.

Note that this same change makes all of Keras a dependency of TensorFlow now, as noted in #22426

**Comment by drasmuss** [comment]
For a concrete example of the impact of these new dependencies, the size of a clean tensorflow installation went from 375MB (1.9.0), to 379MB (1.10.0), to 516MB (1.11.0).

**Comment by WhistleWhileYouWork** [comment]
The updated install docs at: https://www.tensorflow.org/install/source
now state the following as dependencies, but to use the --no-deps flags:
pip install -U --user keras_applications==1.0.5 --no-deps
pip install -U --user keras_preprocessing==1.0.3 --no-deps

**Comment by drasmuss** [comment]
I think we have to consider that the vast majority of users will simply be doing `pip install tensorflow`, so this change makes scipy an install requirement (even if there is technically a way for users to manually work around the standard packaging system behaviour).

Also consider all the downstream packages that have `tensorflow` as a requirement.  They would all now need to add 
```
pip install -U --user keras_applications==1.0.5 --no-deps
pip install -U --user keras_preprocessing==1.0.3 --no-deps
```
to their installation instructions (which would be pretty confusing to users, since those aren't even dependencies of that downstream package) if they want to avoid bloating their own installation process with a scipy dependency.

**Comment by WhistleWhileYouWork** [comment]
Oops, you're right. I inadvertently conflated installing/building from source with simply installing the pre-built package.

**Comment by facaiy** [comment]
It seems to create a little problems because tensorflow depends on keras, see #22601, #22486.  cc @fchollet 

As `tf.keras` has been in tensorflow core, I'm curious about why keras library is still necessary for tf.

**Comment by gunan** [comment]
I think this is resolved for 1.12 and future releases. This dependency was happening through keras_applications.

Thanks for reporting. Unfortunately, for 1.11 we will leave this bug in because we have already started working on 1.12.
